container/image_test.py

```diff
@@ -281,7 +281,7 @@ def test_build_with_tag(self):
 
   def test_with_passwd(self):
     with TestImage('with_passwd') as img:
-      self.assertDigest(img, '9fd1e04ef3070dc98b3b305e9e6091fd9e5fbf4a36d0928b5798a2d9ec2508ba')
+      self.assertDigest(img, 'fd4f1421c53495d773fa189bdcf5e971ed8c28afd92147ad0bb494475c394bb4')
```

Comment:
This concerns me a bit, what changed?

Reply:
So, i exploded the with_passwd.tar and checked them

The with_passwd.runfiles/io_bazel_rules_docker/testdata/with_passwd-layer.tar on the branch and master have same sha.
```
tejaldesai@tejaldesai~/workspace/rules_docker: (fix_passwd_file)$ cat ~/master_with_passwd/with_passwd.runfiles/io_bazel_rules_docker/testdata/with_passwd-layer.tar.sha256 && echo "\n"
41a3e813a70e78f2f50167fc3a4daccf50a73272acc7c54da2211e39473dd2b9\n
          
tejaldesai@tejaldesai~/workspace/rules_docker: (fix_passwd_file)$ cat ~/branch_with_passwd/with_passwd.runfiles/io_bazel_rules_docker/testdata/with_passwd-layer.tar.sha256 && echo "\n"
41a3e813a70e78f2f50167fc3a4daccf50a73272acc7c54da2211e39473dd2b9\n
tejaldesai@tejaldesai~/workspace/rules_docker: (fix_passwd_file)$ 

```
Later they change 
tejaldesai@tejaldesai~/workspace/rules_docker: (fix_passwd_file)$ cat ~/master_with_passwd/with_passwd-layer.tar.sha256 && echo "\n"
f599b8d69097631fa744955945e8a12d0bf130d7cf181564f1d2a9701e1011e3\n
tejaldesai@tejaldesai~/workspace/rules_docker: (fix_passwd_file)$ cat ~/branch_with_passwd/with_passwd-layer.tar.sha256 && echo "\n"
41a3e813a70e78f2f50167fc3a4daccf50a73272acc7c54da2211e39473dd2b9\n
tejaldesai@tejaldesai~/workspace/rules_docker: (fix_passwd_file)$

Reply:
I have no clue why this is happening. but the contents in the tar look exactly the same. I am wondering if the with_passwd.layer.args is coming in play here:
```
 diff ~/master_with_passwd/with_passwd.layer.args ~/branch_with_passwd/with_passwd.layer.args 
4c4
< --tar=bazel-out/local-fastbuild/bin/testdata/testpasswd.passwd.tar
\ No newline at end of file
---
> --tar=bazel-out/local-fastbuild/bin/testdata/passwd_tar.tar
\ No newline at end of file
tejaldesai@tejaldesai~/workspace/rules_docker: (fix_passwd_file)$ 
```

Reply:
so @dlorenc helped me with this and looks like the permissions on the file have changed.
On the branch the permissions are:
```
/branch_with/3b55d28553cf9814bc841d01a71f21f3de32f57bf40a04bc8f19039d4247cb7c:$ ls -l etc/passwd
-r-xr-x--- 1 tejaldesai eng 54 Dec 31  1969 etc/passwd
```
And on the master the permission are 
```
tejaldesai@tejaldesai~/master_with/999fb74aff8830ac7c4c3658d30bef6913f2529ec9a0bc2531ce7b6deb1988b9:$ ls -l etc/passwd 
-rwxr-x--- 1 tejaldesai eng 54 Dec 31  1969 etc/passwd
```
The pkg_impl defaults the mode to 055 here https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/pkg/pkg.bzl#L170.
The initial code did not set any.  I think the right permissions for this file are 0644 i.e 
```
tejaldesai@tejaldesai~:$ ls -l /etc/passwd
-rw-r--r-- 1 root root 2376 Dec  5  2016 /etc/passwd
tejaldesai@tejaldesai~:$ 
```

I am going to update this to make that change. The Sha is still going to change.